src/schedules/commands/get-topics.command.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/database/prisma.service';
+import { GetTopicsQueryParams } from '../dto/get-topics.request.dto';
+import { TopicFactory } from '../utils/topic.fatory';
+import { GetTopicsResponse } from '../dto/get-topics.response.dto';
+import { ScheduleTopicFormatter } from '../utils/schedule-topic-formatter';
+
+@Injectable()
+export class GetTopicsCommand {
+  constructor(private prisma: PrismaService) {}
+
+  async execute(query: GetTopicsQueryParams): Promise<GetTopicsResponse> {
+    let token: string;
+    if (query.name) {
+      token = ScheduleTopicFormatter.formatScheduleTopicToken(query.name);
+    }
+
+    const where = {
+      token: { contains: token },
+    };
+    const take = query.pageSize || 10;
+    const skip = query.page ? (query.page - 1) * take : 0;
+
+    const response = await this.prisma.scheduleTopics.findMany({
+      orderBy: {
+        token: 'asc',
+      },
+      where,
+      skip,
+      take,
+    });
+
+    const totalItems = await this.prisma.scheduleTopics.count({ where });
+
+    return {
+      data: response.map((resp) => TopicFactory.createFromPrisma(resp)),
```

Comment:
Se quiser pode fazer assim pra ficar menos verboso:

```suggestion
      data: response.map(TopicFactory.createFromPrisma),
```